src/bolos/cx_ec.h

```diff
@@ -81,12 +82,165 @@ enum cx_curve_e {
   CX_CURVE_MONTGOMERY_END
   /** High limit (not included) of Montgomery curve ID */
 };
+#else
+#define HAVE_SECP256K1_CURVE
+#define HAVE_SECP256R1_CURVE
+#define HAVE_SECP384R1_CURVE
+#define HAVE_SECP521R1_CURVE
+
+#define HAVE_BRAINPOOL_P256R1_CURVE
+#define HAVE_BRAINPOOL_P256T1_CURVE
+#define HAVE_BRAINPOOL_P320R1_CURVE
+#define HAVE_BRAINPOOL_P320T1_CURVE
+#define HAVE_BRAINPOOL_P384R1_CURVE
+#define HAVE_BRAINPOOL_P384T1_CURVE
+#define HAVE_BRAINPOOL_P512R1_CURVE
+#define HAVE_BRAINPOOL_P512T1_CURVE
+
+#define HAVE_ED25519_CURVE
+#define HAVE_ED448_CURVE
+
+#define HAVE_CV25519_CURVE
+#define HAVE_CV448_CURVE
+
+/** List of supported elliptic curves */
+enum cx_curve_e {
+  CX_CURVE_NONE,
+
+  /* ------------------------ */
+  /* --- Type Weierstrass --- */
+  /* ------------------------ */
+  /** Low limit (not included) of Weierstrass curve ID */
+  CX_CURVE_WEIERSTRASS_START = 0x20,
+
+/** Secp.org */
+#ifdef HAVE_SECP256K1_CURVE
+  CX_CURVE_SECP256K1 = 0x21,
+#define CX_CURVE_256K1 CX_CURVE_SECP256K1
+#endif
+
+#ifdef HAVE_SECP256R1_CURVE
+  CX_CURVE_SECP256R1 = 0x22,
+#define CX_CURVE_256R1    CX_CURVE_SECP256R1
+#define CX_CURVE_NISTP256 CX_CURVE_SECP256R1
+#endif
+
+#ifdef HAVE_SECP384R1_CURVE
+  CX_CURVE_SECP384R1 = 0x23,
+#define CX_CURVE_NISTP384 CX_CURVE_SECP384R1
+#endif
+
+#ifdef HAVE_SECP521R1_CURVE
+  CX_CURVE_SECP521R1 = 0x24,
+#define CX_CURVE_NISTP521 CX_CURVE_SECP521R1
+#endif
+
+#ifdef HAVE_BRAINPOOL_P256T1_CURVE
+  /** BrainpoolP256t1 */
+  CX_CURVE_BrainPoolP256T1 = 0x31,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P256R1_CURVE
+  /** BrainpoolP256r1 */
+  CX_CURVE_BrainPoolP256R1 = 0x32,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P320T1_CURVE
+  /** BrainpoolP320t1 */
+  CX_CURVE_BrainPoolP320T1 = 0x33,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P320R1_CURVE
+  /** BrainpoolP320r1 */
+  CX_CURVE_BrainPoolP320R1 = 0x34,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P384T1_CURVE
+  /** BrainpoolP384t1 */
+  CX_CURVE_BrainPoolP384T1 = 0x35,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P384R1_CURVE
+  /** Brainpool384r1 */
+  CX_CURVE_BrainPoolP384R1 = 0x36,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P512T1_CURVE
+  /** BrainpoolP512t1 */
+  CX_CURVE_BrainPoolP512T1 = 0x37,
+#endif
+
+#ifdef HAVE_BRAINPOOL_P512R1_CURVE
+  /** BrainpoolP512r1 */
+  CX_CURVE_BrainPoolP512R1 = 0x38,
+#endif
+
+  CX_CURVE_BLS12_381_G1 = 0x39,
```

Comment:
Il manque les ifdef HAVE, courbe pas géré sans modification du obj_mac.h.

Reply:
Please open a specific issue for that.